### PR TITLE
Fix 'document.title' property proxing for some cases (close #2328)

### LINKED ISSUE #2328: Latest Docker image breaks updating page titles
### What is your Test Scenario?
I am testing a single-page application. The application uses a default page title (\<head><title>Site</title>\</head>) which is changed via javascript after the page loads.

This only happens when running against the TestCafe Docker image. It doesn't happen in local browsers.

This test just started failing today and I believe it is because the TestCafe Docker image was updated. If I run the tests against testcafe/testcafe:1.8.5, they fail. If I run them against testcafe/testcafe:1.8.4 they pass.
 
### What is the Current behavior?
The page title is never changed from "Site" to "Checkout". It is always "Site".
 
### What is the Expected behavior?
I expect the page title to be "Checkout" after the page is loaded (meaning the JS has successfully updated the title).
 
### What is your web application and your TestCafe test code?

<details>
<summary>Unfortunately the application is private.</summary>

Here is the JavaScript that updates the page title:

```js
document.title = 'Checkout'
```
</details>

<details>
<summary>Your complete test code (or attach your test files):</summary>

 Here is the TestCafe code that checks for the new title:
```js
const pageTitle = Selector(`head title`, { timeout: 10000 })
await t.expect(pageTitle.textContent).contains('Checkout')
```
</details>

<details>
<summary>Your complete configuration file (if any):</summary>

```
{
  "src": "./tests/e2e/specs/*.spec.js",
  "skipJsErrors": true,
  "skipUncaughtErrors": true,
  "concurrency": 4
}
```
</details>

### Steps to Reproduce:
N/A
 
### Your Environment details:

This is all running in the [TestCafe Docker image v1.8.5](https://hub.docker.com/layers/testcafe/testcafe/1.8.5/images/sha256-70d4051fd1fd5cf7a29e346f2eb1438765b156dc82b9d0b4998d631c59b923ba?context=explore).
 
* command-line arguments: --user root 
* browser name and version: fails in both firefox and chromium

### PR DESCRIPTION
**Pull request by miherlosev**
Cases:
* text properties of the first <title> element - (`text`, `textContent`, `innerText`, `innerHTML`)
* edge cases: add/remove `title` elements, element not in DOM, in another window.
* create the first `title` element on `document.title` setter if it's necessary.

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/client/sandbox/node/window.ts [hunk 1]
```diff
@@ -1235,66 +1240,74 @@ export default class WindowSandbox extends SandboxBase {
...
-                            parentTag:        el.tagName,
-                            processedContext: el[INTERNAL_PROPS.processedContext]
-                        });
+                const el = this;
+
+                if (windowSandbox._documentTitleStorage.isAffected(el))
+                    windowSandbox._documentTitleStorage.setTitle(value);
+
+                else {
+                    const isStyleEl  = isStyleElement(el);
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.ts:1249]:
```js
	        if (windowSandbox._documentTitleStorage.isAffected(el)) {
                    windowSandbox._documentTitleStorage.setTitle(value);
                    return el;
                }

                // 'else' is extra there and in other getters and setters

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/client/sandbox/node/document/index.ts [hunk 2]
```diff
@@ -325,9 +326,13 @@ export default class DocumentSandbox extends SandboxBase {
...
-            getter: () => storedTitle,
-            setter: value => {
-                storedTitle = value;
+            getter: function () {
+                return documentSandbox._documentTitleStorage.getTitle();
+            } ,
+            setter: function (value) {
+                documentSandbox._documentTitleStorage.setTitle(value);
+
+                return value;
```
Comment (LavrovArtem) [inline src/client/sandbox/node/document/index.ts:335]: The return statement is unnecessary in the setter of descriptor.

File: src/client/sandbox/node/window.ts [hunk 3]
```diff
@@ -1295,6 +1304,8 @@ export default class WindowSandbox extends SandboxBase {
                     if (isHtmlElement(el) || isBodyElement(el))
                         nativeMethods.setTimeout.call(window, () => windowSandbox.nodeMutation.onBodyContentChanged(el), 0);
                 }
+
+                return value;
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.ts:1308]: same

File: src/client/sandbox/node/window.ts [hunk 4]
```diff
@@ -1340,16 +1351,24 @@ export default class WindowSandbox extends SandboxBase {
...
+                if (windowSandbox._documentTitleStorage.isAffected(this))
+                    windowSandbox._documentTitleStorage.setTitle(value);
+
                 const processedValue = WindowSandbox._processTextPropValue(this, value);
 
                 DOMMutationTracker.onChildrenChanged(this);
 
                 nativeMethods.htmlElementInnerTextSetter.call(this, processedValue);
+
+                return value;
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.ts:1371]: same

File: src/client/sandbox/node/window.ts [hunk 5]
```diff
@@ -1340,16 +1351,24 @@ export default class WindowSandbox extends SandboxBase {
...
+                if (windowSandbox._documentTitleStorage.isAffected(this))
+                    return windowSandbox._documentTitleStorage.getTitle();
+
                 const textContent = nativeMethods.htmlElementInnerTextGetter.call(this);
 
                 return WindowSandbox._removeProcessingInstructions(textContent);
             },
             setter: function (value) {
+                if (windowSandbox._documentTitleStorage.isAffected(this))
+                    windowSandbox._documentTitleStorage.setTitle(value);
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.ts:1363]:
```js
if (windowSandbox._documentTitleStorage.isAffected(this)) {
    windowSandbox._documentTitleStorage.setTitle(value);
    return;
}
```

File: src/client/sandbox/node/window.ts [hunk 6]
```diff
@@ -1383,16 +1402,24 @@ export default class WindowSandbox extends SandboxBase {
...
+
                 const processedValue = WindowSandbox._processTextPropValue(this, value);
 
                 DOMMutationTracker.onChildrenChanged(this);
 
                 nativeMethods.nodeTextContentSetter.call(this, processedValue);
+
+                return value;
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.ts:1422]: same

File: src/client/sandbox/node/window.ts [hunk 7]
```diff
@@ -1485,5 +1512,23 @@ export default class WindowSandbox extends SandboxBase {
...
+                return nativeMethods.titleElementTextGetter.call(this);
+
+            },
+            setter: function (value) {
+                if (windowSandbox._documentTitleStorage.isAffected(this))
+                    windowSandbox._documentTitleStorage.setTitle(value);
+
+                nativeMethods.titleElementTextSetter.call(this, value);
+
+                return value;
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.ts:1530]: same

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Pull request merged**